zcash_primitives/src/transaction/components/amount.rs

```diff
@@ -38,6 +40,14 @@ impl Amount {
         Amount(amount)
     }
 
+    /// Creates a constant Amount from a u64.
+    ///
+    /// Panics: if the amount is outside the range `{0..MAX_MONEY}`.
+    const fn const_from_u64(amount: u64) -> Self {
```

Comment:
Why not `pub`?

Reply:
I want to reconsider these types; `Amount` will become something like `ZatoshiDelta` and be used exclusively in those few cases where negative values make sense, and so I don't want to expose more public APIs. Ideally, we'll eliminate all of the conversions from raw types and make it so that these value deltas can only be produced by subtraction of `Zatoshi` amounts; right now there are some serialization boundaries that still expect i64s but I think that in almost all those cases, the negative part of the range is actually in error.